diff --git a/python/sglang/srt/managers/schedule_batch.py b/python/sglang/srt/managers/schedule_batch.py
index 3ffdd03..702c1c2 100644
--- a/python/sglang/srt/managers/schedule_batch.py
+++ b/python/sglang/srt/managers/schedule_batch.py
@@ -425,7 +425,7 @@ class Req:
             )
             return
 
-        last_token_id = self.output_ids[0]
+        last_token_id = self.output_ids[-1]
 
         matched_eos = False
